routes/dao_coin_exchange.go

```diff
@@ -0,0 +1,183 @@
+
+	ordersBuyingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin1PKID, coin2PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	ordersSellingCoin1, err := adapter.GetAllDAOCoinLimitOrdersForThisDAOCoinPair(coin2PKID, coin1PKID)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDAOCoinLimitOrders: Error getting limit orders: %v", err))
+		return
+	}
+
+	var response []DAOCoinLimitOrder
+
+	for _, order := range ordersBuyingCoin1 {
+		transactorPublicKey := utxoView.GetPublicKeyForPKID(order.TransactorPKID)
+		response = append(response, DAOCoinLimitOrder{
+			TransactorPublicKeyBase58CheckOrUsername: lib.Base58CheckEncode(transactorPublicKey, false, fes.Params),
+
+			BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername:  coin1ProfilePublicBase58Check,
+			SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername: coin2ProfilePublicBase58Check,
+			ScaledExchangeRateCoinsToSellPerCoinToBuy:           order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			ExchangeRateCoinsToSellPerCoinToBuy: floatExchangeRateCoinsToSellPerCoinToBuy(
+				order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			),
+			QuantityToBuyInBaseUnits: order.QuantityToBuyInBaseUnits,
+			QuantityToBuy:            floatQuantityToBuy(order.QuantityToBuyInBaseUnits),
+		})
+	}
+
+	for _, order := range ordersSellingCoin1 {
+		transactorPublicKey := utxoView.GetPublicKeyForPKID(order.TransactorPKID)
+		response = append(response, DAOCoinLimitOrder{
+			TransactorPublicKeyBase58CheckOrUsername: lib.Base58CheckEncode(transactorPublicKey, false, fes.Params),
+
+			BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername:  coin2ProfilePublicBase58Check,
+			SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername: coin1ProfilePublicBase58Check,
+			ScaledExchangeRateCoinsToSellPerCoinToBuy:           order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			ExchangeRateCoinsToSellPerCoinToBuy: floatExchangeRateCoinsToSellPerCoinToBuy(
+				order.ScaledExchangeRateCoinsToSellPerCoinToBuy,
+			),
+			QuantityToBuyInBaseUnits: order.QuantityToBuyInBaseUnits,
+			QuantityToBuy:            floatQuantityToBuy(order.QuantityToBuyInBaseUnits),
+		})
+	}
+
+	_ = json.NewEncoder(ww).Encode(GetDAOCoinLimitOrdersResponse{Orders: response})
+}
+
+// Given a value v, this computes v / (2 ^ 128) and returns it as float
+func floatExchangeRateCoinsToSellPerCoinToBuy(scaledValue *uint256.Int) float64 {
+	base := big.NewInt(2)
+	exponent := big.NewInt(128)
+	divisor := big.NewFloat(0).SetInt(base.Exp(base, exponent, nil))
+
+	scaledValueAsBigFloat := big.NewFloat(0).SetInt(scaledValue.ToBig())
+
+	quotientAsBigFloat := big.NewFloat(0).Quo(
+		scaledValueAsBigFloat,
+		divisor,
+	)
+	quotient, _ := quotientAsBigFloat.Float64()
+	return quotient
+}
+
+// Given a quantity q, this returns q / (NanosPerUnit) as float
+func floatQuantityToBuy(quantityInBaseUnits *uint256.Int) float64 {
```

Comment:
if the return value of this is only used for frontend consumption, maybe we consider return a string? Worried about precision w/ a float64